sdk/src/impl/position-impl.ts

```diff
@@ -93,30 +112,41 @@ export class PositionImpl implements Position {
       throw new Error("Unable to fetch whirlpool for this position.");
     }
 
-    return toTx(
-      this.ctx,
-      decreaseLiquidityIx(this.ctx.program, {
-        ...liquidityInput,
-        whirlpool: this.data.whirlpool,
-        position: this.address,
-        positionTokenAccount: await deriveATA(positionWalletKey, this.data.positionMint),
-        tokenOwnerAccountA: await deriveATA(sourceWalletKey, whirlpool.tokenMintA),
-        tokenOwnerAccountB: await deriveATA(sourceWalletKey, whirlpool.tokenMintB),
-        tokenVaultA: whirlpool.tokenVaultA,
-        tokenVaultB: whirlpool.tokenVaultB,
-        tickArrayLower: PDAUtil.getTickArray(
-          this.ctx.program.programId,
-          this.data.whirlpool,
-          TickUtil.getStartTickIndex(this.data.tickLowerIndex, whirlpool.tickSpacing)
-        ).publicKey,
-        tickArrayUpper: PDAUtil.getTickArray(
-          this.ctx.program.programId,
-          this.data.whirlpool,
-          TickUtil.getStartTickIndex(this.data.tickUpperIndex, whirlpool.tickSpacing)
-        ).publicKey,
-        positionAuthority: positionWalletKey,
-      })
+    const txBuilder = new TransactionBuilder(this.ctx.provider);
+    const [ataA, ataB] = await resolveOrCreateATAs(
```

Comment:
i think one issue with using `resolveOrCreateATAs` is that it makes a network call, and thus makes it harder for clients who want to handle this on their own in a batched manner more difficult

Reply:
Makes sense. Thanks for flagging this. I was talking to @philcchen the other day and I believe after he continues working on transaction packing, he'll work on allowing `resolveOrCreateATAs` to take in an ATA cache. So once that's in, we can expose the cache as a parameter for these types of `WhirlpoolClient` calls.

As of right now, WhirlpoolClient is mainly targeted for simpler projects. From some comments on integration i think it makes more sense to include this for now then add optionality to skip later.  So I suggest we make the trade off here to add the resolve tx while we follow up with the ataMap down the road. Meanwhile, more advanced use cases will just construct their own `decreaseLiquidityIx`

What do you guys think?

Reply:
I'm personally okay with the trade-off for now, and revisiting the batching of calls in the future;

That said, I don't think it would be too difficult to take in a boolean on whether to optionally add the create ATA instructions as a compromise in-between now and then?

Reply:
I've adjusted the methods to allow for optional ATA resolve for decrease-liq only (increase-liq doesn't make sense since you have to have the tokens ready prior to calling).

I also took the chance to fix up the public APIs to support different funder/payer wallet, sourceWallet, destination Wallet for these methods in the following fixes.